platform/pulpcore/app/models/content.py

```diff
@@ -76,39 +45,30 @@ class Artifact(Model):
     Fields:
 
         file (models.FileField): The stored file.
-        downloaded (models.BooleanField): The associated file has been successfully downloaded.
-        requested (models.BooleanField): The associated file has been requested by a client at
-            least once.
-        relative_path (models.TextField): The artifact's path relative to the associated
-            :class:`Content`. This path is incorporated in the absolute storage path of the file
-            and its published path relative to the root publishing directory. At a minimum the path
-            will contain the file name but may also include sub-directories.
         size (models.IntegerField): The size of the file in bytes.
         md5 (models.CharField): The MD5 checksum of the file.
         sha1 (models.CharField): The SHA-1 checksum of the file.
         sha224 (models.CharField): The SHA-224 checksum of the file.
         sha256 (models.CharField): The SHA-256 checksum of the file.
         sha384 (models.CharField): The SHA-384 checksum of the file.
         sha512 (models.CharField): The SHA-512 checksum of the file.
```

Comment:
We may have different validation rules, and I don't know if that might prevent us from having an ABC. For example, on an ``Artifact`` all fields will be required.